product_last_purchase_date/tests/test_product_last_purchase_date.py

```diff
@@ -0,0 +1,92 @@
+                        0,
+                        0,
+                        {
+                            "name": "Test Move",
+                            "location_id": self.location.id,
+                            "location_dest_id": self.location_dest.id,
+                            "product_id": self.product.id,
+                            "product_uom": self.product.uom_id.id,
+                            "product_uom_qty": 1,
+                        },
+                    )
+                ],
+            }
+        )
+        picking.action_confirm()
+        picking.action_assign()
+        picking.button_validate()
+        immediate_wizard = picking.button_validate()
+        self.assertEqual(immediate_wizard.get("res_model"), "stock.immediate.transfer")
+        immediate_wizard_form = Form(
+            self.env[immediate_wizard["res_model"]].with_context(
+                **immediate_wizard["context"]
+            )
+        ).save()
+        immediate_wizard_form.process()
+
+    def test_product_dates_initially_false(self):
+        """Check initial date fields are False."""
+        self.assertFalse(self.product_template.last_purchase_date)
+        self.assertFalse(self.product_template.man_last_purchase_date)
+        self.assertFalse(self.product.last_purchase_date)
+        self.assertFalse(self.product.man_last_purchase_date)
+
+    def test_assign_dates_to_variant(self):
+        """Assign dates to the variant and check propagation to the template."""
+        date = datetime.date.today() + timedelta(days=1)
+        self.product.write(
+            {
+                "last_purchase_date": date,
+                "man_last_purchase_date": date,
+            }
+        )
+        self.assertEqual(self.product_template.last_purchase_date, date)
+        self.assertEqual(self.product_template.man_last_purchase_date, date)
+
+    def test_receipt_updates_dates(self):
+        """Create incoming receipt and check dates update."""
+        current_date = datetime.date.today()
+        self.create_incoming_receipt()
+        self.assertEqual(self.product.last_purchase_date, current_date)
+        self.assertEqual(self.product_template.last_purchase_date, current_date)
+
+        # Ensure 'man_last_purchase_date' later than 'last_purchase_date' updates fields.
+        date = datetime.date.today() + timedelta(days=3)
+        self.product.man_last_purchase_date = date
+        self.assertEqual(self.product.last_purchase_date, date)
+        self.assertEqual(self.product.man_last_purchase_date, date)
+        self.assertEqual(self.product_template.last_purchase_date, date)
+        self.assertEqual(self.product_template.man_last_purchase_date, date)
+
+        # Create picking with current date and check dates do not revert
+        self.create_incoming_receipt()
+        self.assertNotEqual(self.product.last_purchase_date, current_date)
+        self.assertNotEqual(self.product_template.last_purchase_date, current_date)
```

Comment:
We can be more specific with the expected result.
```suggestion
        self.assertEqual(self.product.last_purchase_date, manual_date)
        self.assertEqual(self.product_template.last_purchase_date, manual_date)
```